### PR TITLE
Add locale to query for sitemap-provider

### PR DESCRIPTION
**Pull request by wachterjohannes**

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@wachterjohannes looks good for me can we adopt a test to cover this issue?

File: Content/Infrastructure/Sulu/Sitemap/ContentSitemapProvider.php [hunk 1]
```diff
@@ -81,19 +86,21 @@ public function __construct(
         string $kernelEnvironment,
         string $contentRichEntityClass,
         string $routeClass,
-        string $alias
+        string $alias,
+        int $pageSize = self::PAGE_SIZE
```
Comment (alexander-schranz) [inline Content/Infrastructure/Sulu/Sitemap/ContentSitemapProvider.php]: I personally would no add it to the constructor:

```suggestion
        string $alias
```

File: Content/Infrastructure/Sulu/Sitemap/ContentSitemapProvider.php [hunk 2]
```diff
@@ -81,19 +86,21 @@ public function __construct(
...
     ) {
         $this->entityManager = $entityManager;
         $this->webspaceManager = $webspaceManager;
         $this->kernelEnvironment = $kernelEnvironment;
         $this->contentRichEntityClass = $contentRichEntityClass;
         $this->routeClass = $routeClass;
         $this->alias = $alias;
+        $this->pageSize = $pageSize;
```
Comment (alexander-schranz) [inline Content/Infrastructure/Sulu/Sitemap/ContentSitemapProvider.php]:
```suggestion
        $this->pageSize = $[pageSize](self::PAGE_SIZE);
```

File: Tests/Functional/Content/Infrastructure/Sulu/Sitemap/ContentSitemapProviderTest.php [hunk 3]
```diff
@@ -151,6 +151,11 @@ public function testCreateSitemap(): void
 
     public function testGetMaxPage(): void
     {
+        // set page-size to 5 to check if the query behind it is correct
+        $reflectionProperty = new \ReflectionProperty(\get_class($this->contentSitemapProvider), 'pageSize');
+        $reflectionProperty->setAccessible(true);
+        $reflectionProperty->setValue($this->contentSitemapProvider, 5);
```
Comment (alexander-schranz) [inline Tests/Functional/Content/Infrastructure/Sulu/Sitemap/ContentSitemapProviderTest.php]: you can use the following trait for this: https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/TestBundle/Testing/SetGetPrivatePropertyTrait.php

File: composer.json [hunk 4]
```diff
@@ -12,13 +12,14 @@
...
-        "doctrine/inflector": "^1.4.1 || ^2.0.1",
         "doctrine/collections": "^1.0",
+        "doctrine/doctrine-bundle": "^1.10 || ^2.0",
+        "doctrine/inflector": "^1.4.1 || ^2.0.1",
         "doctrine/orm": "^2.5.3",
         "doctrine/persistence": "^1.3 || ^2.0",
-        "doctrine/doctrine-bundle": "^1.10 || ^2.0",
         "friendsofsymfony/rest-bundle": "^2.6 || ^3.0",
         "massive/search-bundle": "^2.4",
+        "php-http/message-factory": "^1.1",
```
Comment (alexander-schranz) [inline composer.json]: do you know who requires this?

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: composer.json (hunk 4 above)
Comment (wachterjohannes) [reply composer.json]: hmm i only run `composer update` that is realy strange - will revert this

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**